src/keymap.rs

```diff
@@ -913,7 +929,9 @@ impl InputState {
                 }
             }
             KeyPress::Ctrl('J') |
-            KeyPress::Enter => Cmd::AcceptLine,
+            KeyPress::Enter => {
```

Comment:
`AcceptLine == AcceptOrInsertLine { accept_in_the_middle: true }` ?

Reply:
Yes, accept whatever is fully formed command or insert a newline otherwies. That's for backwards compatibility.  As described in the docstring above this is what makes most sense for mostly-single-line cases (i.e. a command-line like in nushell).